components/playback/controls/player/InputDropDown.vue

```diff
@@ -0,0 +1,246 @@
+      :style="'width: ' + input_width + 'px;'"
+    >
+      {{ title_label }}
+      <!--  original mockflow ID: cmpDb072c1da7a823374cbee04cb1666edb1   -->
+    </span>
+
+    <div
+      class="div__input-dropdown-controls-content-widget"
+      :class="[
+        message_if_blank
+          ? 'div__input-dropdown-controls-content-widget--invalid'
+          : 'div__input-dropdown-controls-content-widget--valid',
+      ]"
+      :style="
+        'width: ' + input_width + 'px;' + 'top:' + input_widget_top + 'px;'
+      "
+    >
+      <span
+        class="span__input-dropdown-controls-content-input-txt-widget"
+        :style="'width: ' + input_width + 'px;'"
+      >
+        <b-form-input
+          id="input-dropdown-widget"
+          v-model="input_dropdown_value_key"
+          list="option_list"
+          :placeholder="placeholder"
+          :disabled="disabled"
+          class="w-100 h-100 edit-id"
+          style="
+            border-radius: 0;
+            background-color: #1c1c1c;
+            border: 0px;
+            color: #ffffff;
+          "
+        ></b-form-input>
+        <datalist v-if="dropdown_options.length" id="option_list">
+          <option v-for="item in dropdown_options" :id="item.id" :key="item.id">
+            {{ item.name }}
+          </option>
+        </datalist>
+      </span>
+    </div>
+    <div
+      v-show="message_if_blank"
+      class="div__input-dropdown-controls-content-feedback"
+      :style="
+        'width: ' + input_width + 'px;' + 'top:' + input_feedback_top + 'px;'
+      "
+    >
+      {{ invalid_text }}
+    </div>
+  </div>
+</template>
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import { BFormInput } from "bootstrap-vue";
+Vue.use(BootstrapVue);
+Vue.component("BFormInput", BFormInput);
+import "bootstrap/dist/css/bootstrap.min.css";
+export default {
+  name: "InputDropDown",
+  props: {
+    title_label: { type: String, default: "" }, // title_text (str) (optional, defaults to empty string "")
+    placeholder: { type: String, default: "" }, // placeholder (str)
+    invalid_text: { type: String, default: "" }, // invalid_text (str)
+    value: { type: String, default: "" }, // field_value (str) (optional, defaults to empty string "")
+    input_width: { type: Number, default: 0 }, // textbox_width (int)  [pixels]
+    disabled: { type: Boolean, default: false }, // disabled (optional bool=False) (not able to type into input)
+    options_text: { type: Array, required: true },
+    message_if_blank: { type: Boolean, default: false }, // when set to true, will display a simple feedback
+  },
+  data() {
+    return {
+      input_dropdown_value_key: this.value,
+      input_width_background: this.input_width + 4,
+    };
+  },
+  computed: {
+    input_key_action: function () {
+      return this.message_if_blank;
+    },
+    dropdown_options: function () {
+      if (this.options_text.length == 0) {
+        return [];
```

Comment:
please make sure this line/logic is tested. In test driven development, we don't write a line of code without first having a failing test.   You could test it as part of a test `Given the component is mounted with no options, When the prop options_text is updated by the parent and the user types one of the new valid options, Then an update event is emitted with the typed option`.   This then also tests to confirm the component reacts appropriately to a prop update---since there is a lot of logic between updating that prop and the desired final result of getting the items into the `datalist` with the right ids, it's not a bad idea to confirm it is all wired up correctly.

Reply:
Agreed will try to introduce the unit test to validate the sequence.